frontend/javascripts/oxalis/geometries/materials/plane_material_factory.ts

```diff
@@ -455,6 +489,35 @@ class PlaneMaterialFactory {
         true,
       ),
     );
+    this.storePropertyUnsubscribers.push(
+      listenToStoreProperty(
+        (storeState) => storeState.dataset.dataSource.dataLayers,
+        (layers) => {
+          let representativeLayerIdxForMag = null;
+          for (let layerIdx = 0; layerIdx < layers.length; layerIdx++) {
+            const layer = layers[layerIdx];
+            const name = sanitizeName(layer.name);
+            this.uniforms[`${name}_transform`].value = invertAndTranspose(
+              layer.transformMatrix || Identity4x4,
+            );
+            const hasTransform = !_.isEqual(layer.transformMatrix || Identity4x4, Identity4x4);
+            this.uniforms[`${name}_has_transform`] = {
+              value: hasTransform,
+            };
+            if (representativeLayerIdxForMag == null && !hasTransform) {
+              representativeLayerIdxForMag = layerIdx;
+            }
+          }
+
+          // If all layers have a transform, the representativeLayerIdxForMag
```

Comment:
Could you briefly explain what the `representativeLayerIdxForMag` is needed/used for?

Reply:
Good point! I added the following comment:

> // The vertex shader looks up the buckets for rendering so that the
// fragment shader doesn't need to do so. Currently, this only works
// for layers that don't have a transformation (otherwise, the differing
// grids wouldn't align with each other).
// To align the vertices with the buckets, the current magnification is
// needed. Since the current mag can differ from layer to layer, the shader
// needs to know which mag is safe to use.
// For this purpose, we define the representativeLayerIdxForMag which is
// essentially an index to *some* layer that doesn't have any transforms.
// If all layers have a transform, the representativeLayerIdxForMag
// isn't relevant which is why it can default to 0.